Marlin/src/HAL/HAL_LPC1768/main.cpp

```diff
@@ -90,7 +90,7 @@ void HAL_init() {
   delay(1000);                              // Give OS time to notice
   USB_Connect(TRUE);
 
-  #if DISABLED(USB_SD_DISABLED)
+  #if DISABLED(USB_SD_DISABLED) && !BOTH(SHARED_SD_CARD, INIT_SDCARD_ON_BOOT)
```

Comment:
Note that the option `INIT_SDCARD_ON_BOOT` refers to the SD card used for printing which is initialized by `card.init`.

Now, I see here you're testing `BOTH(SHARED_SD_CARD, INIT_SDCARD_ON_BOOT)` and expecting something to change. It never will. `INIT_SDCARD_ON_BOOT` will never be enabled if `SHARED_SD_CARD` is enabled by your own added logic:
```cpp
#if ENABLED(SDSUPPORT) && DISABLED(SHARED_SD_CARD)
  #define INIT_SDCARD_ON_BOOT
#endif
```
So this only needs to check for `INIT_SDCARD_ON_BOOT` having been set at all.

Reply:
I'm allowing for users to force INIT_SDCARD_ON_BOOT by explicitly adding it into their config. Some users may want to always have the SD card mounted on boot even when sharing it via USB, this option may cause problems with some hosts (like a Mac) so I would not suggest putting it into the configuration files, but I think it makes sense to allow for it.

That was one of the configurations I tested and on Windows/Linux it works well.

Reply:
We are going to need more testing of `INIT_SDCARD_ON_BOOT` as an override than just your limited setup. And that is the standard transparent behavior for all single-SD setups, so perhaps the more appropriate option would be `DONT_MOUNT_SD_ON_BOOT` but also we need to make clearer which SD we are talking about.

Why would you think that macOS (a fairly standard UNIX) would have host drive issues?

Reply:
Yes perhaps DONT_MOUNT_SD_ON_BOOT would be the most appropriate setting. As I said above I am not suggesting that this option be provided as a standard option in the configuration files, but I see no harm whatsoever in having the code protect against it. At the moment if a user sets it you will end up with both the USB code and the Marlin code accessing the SD card at the same time, and I don't think that is a good idea. 

As to why do I think that macOS has a problem, did you read the issue that started all of this (and was referenced in this PR)? That is a report by two (macOS) users that have uncounted problems following the change to always mount SD cards on boot. For whatever reason that OS seems to handle this type of USB device differently to others. It may be the automounting logic that is the problem but who knows. I have no way to investigate further, I do know that two different versions of Windows and three different versions of Linux have no problems, but that is the extent of my "limited setup" to test.